app/src/main/java/org/gdg/frisbee/android/arrow/ArrowActivity.java

```diff
@@ -249,65 +250,107 @@ private void score(final String id) {
-                                Toast.makeText(ArrowActivity.this, R.string.arrow_already_tagged, Toast.LENGTH_LONG).show();
-                            } else {
-                                addTaggedPersonToCloudSave(id);
-                            }
-                        } else {
-                            addTaggedPersonToCloudSave(id);
-                        }
-                    }
-                } else if (conflictResult != null) {
-                    previous = mergeIds(new String(conflictResult.getLocalData()), new String(conflictResult.getServerData()));
+        new StoreSnapshotTask(getGoogleApiClient()).execute(id);
+    }
 
-                    if (previous.contains(id)) {
-                        Toast.makeText(ArrowActivity.this, R.string.arrow_already_tagged, Toast.LENGTH_LONG).show();
-                    } else {
-                        addTaggedPersonToCloudSave(id);
-                    }
+    public class StoreSnapshotTask extends AsyncTask<String, Void, String> {
 
-                    Toast.makeText(ArrowActivity.this, getString(R.string.arrow_oops), Toast.LENGTH_LONG).show();
+        final GoogleApiClient googleApiClient;
+
+        public StoreSnapshotTask(GoogleApiClient googleApiClient) {
+            this.googleApiClient = googleApiClient;
+        }
+
+        @Override
+        protected String doInBackground(String... params) {
+            String id = params[0];
+
+            try {
+                Snapshot snapshot = openSnapshot();
+                if (snapshot != null) {
+                    storeInSnapshot(snapshot, id);
+
+                    String personName = getTaggedPersonName(id);
+                    return "It worked...you tagged " + personName;
                 }
+            } catch (IOException e) {
+                Timber.w(e, "Could not store tagged organizer");
+            } catch (AlreadyTaggedException e) {
+                return getString(R.string.arrow_already_tagged);
             }
-        });
-    }
+            return null;
+        }
 
-    private String mergeIds(String list1, String list2) {
-        String[] parts1 = list1.split(ID_SEPARATOR_FOR_SPLIT);
-        String[] parts2 = list2.split(ID_SEPARATOR_FOR_SPLIT);
-        Set<String> mergedSet = new HashSet<>(Arrays.asList(parts1));
-        mergedSet.addAll(Arrays.asList(parts2));
-        return TextUtils.join(ID_SPLIT_CHAR, mergedSet);
-    }
+        @Override
+        protected void onPostExecute(String message) {
 
-    private void addTaggedPersonToCloudSave(String id) {
-        previous = previous + ID_SPLIT_CHAR + id;
-        AppStateManager.update(getGoogleApiClient(), Const.ARROW_DONE_STATE_KEY, previous.getBytes());
-        Games.Leaderboards.submitScore(getGoogleApiClient(), Const.ARROW_LB, previous.split("\\|").length - 1);
+            if (message != null) {
+                Toast.makeText(ArrowActivity.this, message, Toast.LENGTH_LONG).show();
```

Comment:
Yes, this will probably cause problems. I will do something about it before release.